apps/jubmoji-quest/src/pages/api/quests.ts

```diff
@@ -0,0 +1,117 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "@/lib/prisma";
+import { getRandomNullifierRandomness } from "jubmoji-api";
+import { bigIntToHex } from "babyjubjub-ecdsa";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
+    try {
+      const quests = await prisma.quest.findMany({
+        select: {
+          id: true,
+          name: true,
+          description: true,
+          tags: true,
+          startTime: true,
+          endTime: true,
+          proofType: true,
+          proofParams: true,
+          // Selecting specific fields from related models
+          prerequisiteCards: {
+            select: {
+              index: true,
+            },
+          },
+          collectionCards: {
+            select: {
+              index: true,
+            },
+          },
+        },
+      });
+
+      const transformedQuests = quests.map((quest) => {
+        const prerequisiteIndices = quest.prerequisiteCards.map(
+          (card) => card.index
+        );
+        const collectionIndices = quest.collectionCards.map(
+          (card) => card.index
+        );
+        return {
+          ...quest,
+          prerequisiteCardIndices: prerequisiteIndices,
```

Comment:
Why do we do this index processing here instead of on the frontend? Should cleanup this API code a bunch?

Reply:
I think it makes more sense to do this transformation on the backend since when we actually define the types for the frontend, they will only use cardIndices rather than the relation itself. I will formalize the frontend type definitions in my next changes connecting the data store to the frontend